File: problems/easy/grading_students/grading_students.js
/**
 * Rounds the grades according to the rules.
 * @param {number[]} grades - array of student grades
 * @returns {number[]} - array of rounded grades
 */

function gradingStudents(grades) {
    return grades.map(grade => {
        if (grade < 38) {
            return grade; // no rounding if failing anyway
        }
        const nextMultipleOf5 = Math.ceil(grade / 5) * 5;
        if (nextMultipleOf5 - grade < 3) {
            return nextMultipleOf5; // round up
        }
        return grade; // no rounding
    });
}

module.exports = { gradingStudents };

File: problems/easy/grading_students/grading_students.c
#include <stdio.h>
#include <stdlib.h>

/**
 * Rounds the grades according to the rules.
 * @param grades array of student grades
 * @param n size of array
 * @param result output array to store rounded grades
 */

int* gradingStudents(int grades[], int grades_count) {
    int* result = (int*)malloc(sizeof(int) * grades_count);
    for (int i = 0; i < grades_count; i++) {
        result[i] = grades[i];
    }
    return result;
}
